server/auth.go

```diff
@@ -531,6 +530,42 @@ func (s *Server) isGatewayAuthorized(c *client) bool {
 	if opts.Gateway.Username == "" {
 		return true
 	}
+
+	// Check whether TLS map is enabled, otherwise use single user/pass.
+	if opts.Gateway.TLSMap {
+		tlsState := c.GetTLSConnectionState()
```

Comment:
Make a common func.

Reply:
will add the common func for routes/gateway then update with the squashed commits